### ISSUE TITLE
Implicit option names should be kebabed

### ISSUE THREAD

**Issue by SUPERCILEX** [gradle/gradle#7400]
Example:
```kt
@Option(description = "Trigger default templates update")
@get:Input
var updateTemplates: Boolean = false
```

The option name becomes `updateTemplates` but it should be `update-templates`.

**Comment by eriwen** [comment]
This does seem like a breaking change (if minor), so in my opinion it should be Gradle 6.0. Possibly `name` should be required. 

I'd say "a contribution is welcome" but it would be kinda weird to have to save a PR for nearly a year to merge it.

**Comment by SUPERCILEX** [comment]
Yeah, that's what I was thinking. 👍 Well, I'm already excited for Gradle 6! 😆

**Comment by big-guy** [comment]
We could have both options available and emit a deprecation warning for the non-kebab names.

We should update the docs to say that specifying an `option` value is a good practice for setters/fields that have names that should be kebabed. This is what we're doing in `InitBuild`.

**Comment by CristianGM** [comment]
This was waiting for Gradle 6.0 to do a small breaking change, now this train seems to be gone..

**Comment by Vampire** [comment]
Would still be nice to have, not present in 6.7.1.
7.0 is around the corner, so maybe now would be a good time for the breaking change? :-)
